nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/StandardProvenanceReporter.java

```diff
@@ -445,12 +445,27 @@ public void modifyAttributes(final FlowFile flowFile) {
         modifyAttributes(flowFile, null);
     }
 
+    @Override
+    public void modifyAttributes(final FlowFile flowFile, final long processingMills) {
+        modifyAttributes(flowFile, null, -1L);
+        verifyFlowFileKnown(flowFile);
+    }
+
     @Override
     public void modifyAttributes(final FlowFile flowFile, final String details) {
+        modifyAttributes(flowFile, details, -1L);
+        verifyFlowFileKnown(flowFile);
```

Comment:
Same comment as above; `modifyAttributes(flowFile, details, -1L);` should invoke the method on line 461, which calls `verifyFlowFileKnown(flowFile);` also.

Reply:
Removed same.